changes/bug25152

```diff
@@ -0,0 +1,8 @@
+  o Major bugfixes (performance):
+    - Avoid unnecessary calls to circuitmux_find_map_entry() by caching
+      chanid_circid_muxinfo_t entries as much as possible. This is done
+      by passing in chanid_circid_muxinfo_t into functions in circuitmux.c
+      such as circuitmux_make_circuit_active() where a hashmap entry stays
+      the same across multiple subsequent function calls. Fixes bug 25152;
+      bugfix on 0.3.5.1-alpha. Patch by Neel Chauhan.
```

Comment:
When we say "Bugfix on X", we mean that we are fixing a bug, and the bug first appeared in version X.  So if this performance issue first appeared in version 0.a.b.c, and we're fixing it in 0.x.y.z, then we should say "Bugfix on 0.a.b.c", not "Bugfix on 0.x.y.z". 

I think we could just as easily call this particular one a feature.